dcm4che-conf/dcm4che-conf-core/src/test/java/org/dcm4che3/conf/core/refindexing/RefIndexingTest.java

```diff
@@ -65,6 +64,45 @@ public void testSimpleChanges() {
 
     }
 
+    @Test
+    public void refreshNode_whenNodeIsNotAvailableInConfigurationAnymore_removesNodeFromIndex() {
+        SampleBigConf sampleBigConf = new SampleBigConf();
+
+        sampleBigConf.child1 = new SampleReferableConfClass("UUID1");
+        sampleBigConf.child1.setMyName("Romeo");
+
+        SampleReferableConfClass uuid2obj = new SampleReferableConfClass("UUID2");
+        uuid2obj.setMyName("Juliet");
+
+        sampleBigConf.childList.add(uuid2obj);
+        sampleBigConf.childList.add(new SampleReferableConfClass("UUID3"));
+
+        sampleBigConf.childMap.put("first", new SampleReferableConfClass("UUID4"));
+        sampleBigConf.childMap.put("second", new SampleReferableConfClass("UUID5"));
+
+        configuration.persistNode(new Path("confRoot"), vitalizer.createConfigNodeFromInstance(sampleBigConf), null);
+
+        // check if uuids are indexed
+        Assert.assertEquals(5, uuidToSimplePathCache.size());
+
+        PathPattern pathPattern = new PathPattern(Configuration.REFERENCE_BY_UUID_PATTERN);
+
+        Assert.assertEquals("Romeo", ((Map) configuration.getConfigurationNode(configuration.getPathByUUID("UUID1"), null)).get("myName"));
+        Assert.assertEquals("Juliet", ((Map) configuration.getConfigurationNode(configuration.getPathByUUID("UUID2"), null)).get("myName"));
+
+        Path pathToRemove = new Path("confRoot", "childMap");
+        // cache the node first
+        configuration.getConfigurationNode(pathToRemove, null);
+        lowerLevelConfiguration.removeNode(pathToRemove);
+        configuration.refreshNode(pathToRemove);
+
+        Assert.assertFalse(uuidToSimplePathCache.containsKey("UUID4"));
```

Comment:
You should use allOf and hasKey Matchers together with assertThat to do this as one assert,

Reply:
Done